### PR TITLE
feat: add namespace binding

### PR DESCRIPTION
**Pull request by shubham-cmyk**
## Explanation

Add $namespace to the binding

## Related issue

Fixes https://github.com/kyverno/chainsaw/issues/819

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/kyverno/blob/main/CONTRIBUTING.md).
- [x] I have read the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) and followed the process including adding proof manifests to this PR.
- [ ] This is a bug fix and I have added unit tests that prove my fix is effective.

### CODE REVIEW

File: pkg/runner/processors/test.go [hunk 1]
```diff
@@ -120,6 +121,11 @@ func (p *testProcessor) Run(ctx context.Context, nspacer namespacer.Namespacer)
 			}
 			namespace = &ns
 		}
+		if namespace != nil {
+			bindings = p.bindings.Register("$namespace", binding.NewBinding(namespace.Name))
+		} else {
+			bindings = p.bindings.Register("$namespace", binding.NewBinding(""))
+		}
```
Comment (eddycharly) [inline pkg/runner/processors/test.go:128]: What if `nspacer` is not nil ?

**Comment by eddycharly** [comment]
Why not just ?
```golang
if nspacer != nil{
  bindings = p.bindings.Register("$namespace", binding.NewBinding(nspacer.GetNamespace()))
}
```

**Comment by shubham-cmyk** [comment]
@eddycharly I am thinking this should change with the resource too.

**Comment by eddycharly** [comment]
What ?

**Comment by shubham-cmyk** [comment]
@eddycharly 
if i specify a  namespace in the resource

**Comment by eddycharly** [comment]
> so if a resource has a namespace already then it won't change it

correct

**Comment by shubham-cmyk** [comment]
@eddycharly I am fixing the test

File: .release-notes/main.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ Release notes for `TODO`.
 ## 💫 New features 💫
 
 - Allowed passing test folders by args (`chainsaw test ./folder` instead of `chainsaw test --test-dir ./folder`)
+- Add $namespace binding to check
```
Comment (eddycharly) [inline .release-notes/main.md:24]:
```suggestion
- Added new binding `$namespace` containing the test namespace name
```

**Review by eddycharly — APPROVED** [review]

**Pull request merged**